AudioKit/Common/Internals/AudioKit.swift

```diff
@@ -28,7 +28,31 @@ public typealias AKCallback = () -> Void
     // MARK: - Internal audio engine mechanics
 
     /// Reference to the AV Audio Engine
-    @objc open static var engine = AVAudioEngine()
+    @objc open static var engine = AVAudioEngine() {
+        didSet {
+            #if os(iOS)
+            if routeChangeObserver != nil { NotificationCenter.default.removeObserver(routeChangeObserver!) }
+            // Subscribe to route changes that may affect our engine
+            // Automatic handling of this change can be disabled via AKSettings.enableRouteChangeHandling
+            routeChangeObserver = NotificationCenter.default.addObserver(forName: .AVAudioSessionRouteChange,
+                                                                         object: nil,
+                                                                         queue: OperationQueue.main,
+                                                                         using: { (notification) in
+                                                                            AudioKit.restartEngineAfterRouteChange(notification)
+            })
+
+            if configurationChangeObserver != nil { NotificationCenter.default.removeObserver(routeChangeObserver!) }
```

Comment:
Use if let here too.
Are configurationChangeObserver and routeChangeObserver being conflated here?